### PR TITLE
SDN-3733,SDN-3507,OCPBUGS-16767: [DownstreamMerge] 7-27-23

### PR DESCRIPTION
**Pull request by trozet**
Important fixes:
- Logging fixes to reduce amount of logging in services controller
- Cluster manager leader lock changed to legacy name to avoid duplicate leaders during 4.13->4.14 upgrade
- Fixes race where a per node LB might be temporarily removed and cause disruption during upgrade

@flavio-fernandes @skitt @dcbw @tssurya @ricky-rav

### CODE REVIEW

File: Dockerfile.base [hunk 1]
```diff
@@ -13,7 +13,7 @@ RUN dnf install -y --nodocs \
 	dnf clean all
 
 ARG ovsver=3.1.0-32.el9fdp
-ARG ovnver=23.06.0-13.el9fdp
+ARG ovnver=23.06-23.06.0-51.el9fdp
```
Comment (flavio-fernandes) [inline Dockerfile.base:16]: @trozet:  Is this downstream only? 
That is the only discrepancy I noticed while mocking a ds merge to compare with this PR.

```
❯ git diff tim/merge-7-27-23
diff --git a/Dockerfile.base b/Dockerfile.base
index f7433f928..54a2e033e 100644
--- a/Dockerfile.base
+++ b/Dockerfile.base
@@ -13,7 +13,7 @@ RUN dnf install -y --nodocs \
        dnf clean all

 ARG ovsver=3.1.0-32.el9fdp
-ARG ovnver=23.06-23.06.0-51.el9fdp
+ARG ovnver=23.06.0-13.el9fdp

 RUN INSTALL_PKGS="iptables" && \
        ovsver_short=$(echo "$ovsver" | cut -d'.' -f1,2) && \

```

Comment (trozet) [reply Dockerfile.base:16]: yah to pick up the ct zone fix

**Comment by tssurya** [comment]
hmm https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ovn-kubernetes/1789/pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-shared-to-local-gateway-mode-migration/1684947536906817536 shows similar symptoms - takes longer than 5mins for masters to roll out, i have a bug tracking that here: https://issues.redhat.com/browse/OCPBUGS-16629 but that is supposed to be on IC ROFL, wonder why it takes long on nonIC.. maybe glitch, will retest and see

**Comment by tssurya** [comment]
/retitle SDN-3733,SDN-3507,OCPBUGS-16767: [DownstreamMerge] 7-27-23

**Comment by tssurya** [comment]
hcp also seems unrelated?
```
E0728 18:05:31.138439    2235 memcache.go:265] couldn't get current server API group list: Get "https://ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com:6443/api?timeout=32s": dial tcp: lookup ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com on 172.30.0.10:53: no such host
E0728 18:05:31.139450    2235 memcache.go:265] couldn't get current server API group list: Get "https://ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com:6443/api?timeout=32s": dial tcp: lookup ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com on 172.30.0.10:53: no such host
E0728 18:05:31.140443    2235 memcache.go:265] couldn't get current server API group list: Get "https://ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com:6443/api?timeout=32s": dial tcp: lookup ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com on 172.30.0.10:53: no such host
E0728 18:05:31.141437    2235 memcache.go:265] couldn't get current server API group list: Get "https://ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com:6443/api?timeout=32s": dial tcp: lookup ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com on 172.30.0.10:53: no such host
Unable to connect to the server: dial tcp: lookup ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com on 172.30.0.10:53: no such host
2023-07-28 18:05:31+00:00 Clusteroperators not yet ready
E0728 18:06:02.280475    2275 memcache.go:265] couldn't get current server API group list: Get "https://ac3e69869487c43ce993e647b06fd91a-32c508ba754a6352.elb.us-east-1.amazonaws.com:6443/api?timeout=32s": dial tcp 52.7.169.58:6443: i/o timeout
```

File: Dockerfile.base [hunk 2]
```diff
@@ -13,7 +13,7 @@ RUN dnf install -y --nodocs \
 	dnf clean all
 
 ARG ovsver=3.1.0-32.el9fdp
-ARG ovnver=23.06.0-13.el9fdp
+ARG ovnver=23.06.0-51.el9fdp
```
Comment (flavio-fernandes) [inline Dockerfile.base:16]: Note: 
https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=2614540
look at changelog to see what changed between -13 and -51

**Pull request merged**